pandas/core/algorithms.py

```diff
@@ -547,10 +566,12 @@ def mode(values):
         constructor = Series
 
     dtype = values.dtype
-    if is_integer_dtype(values):
+    if is_signed_integer_dtype(values):
         values = _ensure_int64(values)
         result = constructor(sorted(htable.mode_int64(values)), dtype=dtype)
```

Comment:
side note, not sure whey ``sorted`` is used here, this does a conversion to list (then back when it is constructed).

can you do separate PR to fix this? (use np.sort)